### ISSUE TITLE
start_stop_with_containerd_runtime: kubeadm preflight: docker service is not active

### ISSUE THREAD

**Issue by tstromberg** [kubernetes/minikube#3422]
Seen at go/bugbear/logs/OSX_Integration_Tests_hyperkit/builds/1587

```
E1206 04:22:12.928778 5781 start.go:332] Error starting cluster: kubeadm init error 
sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests --ignore-preflight-errors=DirAvailable--data-minikube --ignore-preflight-errors=Port-10250 --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml --ignore-preflight-errors=FileAvailable--etc-kubernetes-manifests-etcd.yaml --ignore-preflight-errors=Swap --ignore-preflight-errors=CRI --ignore-preflight-errors=Service-Docker --ignore-preflight-errors=Port-8443 --ignore-preflight-errors=Port-10251 --ignore-preflight-errors=Port-10252 --ignore-preflight-errors=Port-2379 
...
, output: [init] using Kubernetes version: v1.12.3
[preflight] running pre-flight checks
[WARNING CRI]: container runtime is not running: output: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
, error: exit status 1
[WARNING Service-Docker]: docker service is not active, please run 'systemctl start docker.service'
[WARNING Swap]: running with swap on is not supported. Please disable swap
[preflight] The system verification failed. Printing the output from the verification:
[0;37mKERNEL_VERSION[0m: [0;32m4.15.0[0m
...
[0;37mCGROUPS_CPUSET[0m: [0;32menabled[0m
[0;37mCGROUPS_DEVICES[0m: [0;32menabled[0m
[0;37mCGROUPS_FREEZER[0m: [0;32menabled[0m
[0;37mCGROUPS_MEMORY[0m: [0;32menabled[0m
[preflight] Some fatal errors occurred:
[ERROR SystemVerification]: failed to get docker info: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
...
map[string]interface {}{"wantreporterrorprompt":false, "ingress":false}
Defaults:
map[string]interface {}{"wantreporterror":false, "showdriverdeprecationnotification":true, "showbootstrapperdeprecationnotification":true, "v":"0", "log_dir":"", "wantupdatenotification":true, "reminderwaitperiodinhours":24, "wantnonedriverwarning":true, "alsologtostderr":"false", "wantreporterrorprompt":true, "wantkubectldownloadmsg":true}
Starting local Kubernetes v1.12.3 cluster...
...
Setting up kubeconfig...
Stopping extra container runtimes...
Restarting containerd runtime...
Starting cluster components...
FAIL
```

**Comment by priyawadhwa** [comment]
This bug was introduced with kubernetes v1.12, which currently doesn't work with containerd in minikube.  The preflight error happens because the `SystemVerification` check makes sure that docker is running. v1.12 uses this [version](https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/kubeadm/templates.go#L48) of the kubeadm template which has the [criSocket](https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/kubeadm/templates.go#L63) harcoded as the docker socket, so the system verification check always fails with containerd.

We would have to remove this hardcoding to get containerd to work with v1.12 when we re-upgrade to 1.12

**Comment by afbjorklund** [comment]
This was supposed to get the proper value of CRISocket, with docker socket only being the default...

See https://github.com/kubernetes/minikube/blob/master/pkg/minikube/bootstrapper/kubeadm/kubeadm.go#L270

Maybe you forgot to use `--container-runtime=containerd` ?

https://github.com/kubernetes/minikube/blob/master/docs/alternative_runtimes.md#using-containerd

**Comment by afbjorklund** [comment]
Think the bug was already in `v1.11`, reported in #3153 fixed in #3154
